test/extended/util/workloads.go

```diff
@@ -75,48 +75,52 @@ func NewDeployment(opts ...DeployOption) Deployment {
 
 // Create the deployment
 func (dep *Deployment) Create(k *CLI) {
-	err := k.WithoutNamespace().WithoutKubeconf().Run("create").Args("deployment", dep.name, "-n", dep.namespace, "--image", dep.image).Execute()
+	err := k.WithoutNamespace().WithoutKubeconf().Run("create").Args("deployment", dep.Name, "-n", dep.Namespace, "--image", dep.Image).Execute()
 	o.Expect(err).NotTo(o.HaveOccurred())
 }
 
 // Delete the deployment
 func (dep *Deployment) Delete(k *CLI) {
-	err := k.WithoutNamespace().WithoutKubeconf().Run("delete").Args("deployment", dep.name, "-n", dep.namespace).Execute()
-	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(dep.Clean(k)).NotTo(o.HaveOccurred())
+}
+
+// Clean the deployment resource
+func (dep *Deployment) Clean(k *CLI) error {
+	return k.WithoutNamespace().WithoutKubeconf().Run("delete").Args("deployment", dep.Name, "-n", dep.Namespace).Execute()
 }
 
 // GetFieldByJSONPath gets specific field value of the deployment by jsonpath
 func (dep *Deployment) GetFieldByJSONPath(k *CLI, JSONPath string) (string, error) {
-	return k.WithoutNamespace().WithoutKubeconf().Run("get").Args("deployment", dep.name, "-n", dep.namespace, "-o", "jsonpath="+JSONPath).Output()
+	return k.WithoutNamespace().WithoutKubeconf().Run("get").Args("deployment", dep.Name, "-n", dep.Namespace, "-o", "jsonpath="+JSONPath).Output()
```

Comment:
@Phaow I already added LGTM, but one more comment you can consider to update in separate PR: for SyncTarget's methods you always use `Args("--server="+ws.ParentServerURL`, which is needed in the context of parallel execution. For Deployment, should it always use too for the sake of parallel execution?